veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,345 @@
+It comprises:
+- A [sidecar container](#sidecar-container) running in the virt-launcher pod. Its sole responsibility is to mutate the DomainXML.
+- A [CNI Plugin](#cni-plugin) that is invoked during pod setup/teardown. 
+- Configuration.
+Further details can be found in the [design document](https://github.com/kubevirt/community/blob/main/design-proposals/network-binding-plugin/network-binding-plugin.md#design-details).
+
+#### Sidecar container
+
+`vhost-user` enables high-performance communication between virtual machines and user-space processes, typically offloading VirtIO emulation to dedicated user-space processes (e.g., DPDK or passt). 
+It improves I/O performance through zero-copy mechanisms and reduced context-switching overhead. Additionally, it is required for seamless TCP migration.
+
+`vhost-user` must be enabled in the DomainXML. Small code changes are required in passt sidecar [code](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/sidecars/network-passt-binding). The format is described [here](https://libvirt.org/formatdomain.html#vhost-user-connection-with-passt-backend).
+
+#### CNI plugin
+
+The [passt-cni](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/cniplugins/passt-binding) process is called by `Multus` during primary network interface setup/removal. It runs two sysctl commands:
+1. Allow binding to all ports starting from 0.
+2. Set ping group range to virt-launcher's user ID.
+> [!NOTE] 
+> No changes are expected in the `passt` CNI code.
+
+### Build and Deployment
+
+#### Add passt-repair binary to virt-handler image
+
+The `passt-repair` binary will be added to the virt-handler container image installed from the `passt` RPM.
+To maintain storage efficiency and eliminate unnecessary security risks, the `passt-repair` binary will be extracted from the `passt` RPM and installed in the virt-handler container image without the `passt` binary.
+
+#### passt CNI plugin build and deployment manifests
+
+The [passt-binding directory](https://github.com/kubevirt/kubevirt/tree/release-1.5/cmd/cniplugins/passt-binding) currently includes a build system to create the CNI plugin binary.
+A few artifacts are required for the deployment of the network binding plugin:
+- A container image build and publish process for the sidecar container.
+- A static daemonset manifest which serves to deploy the CNI.
+- A static manifest for applying a `network-attachment-definition` CR named `netbindingpasst` referencing to the CNI.
+[CNAO](https://github.com/kubevirt/cluster-network-addons-operator) will be enhanced to store and automate the deployment process.
+
+## API Examples
+
+No API changes are expected in kubevirt/kubevirt.
+### passt network binding plugin configuration in KubeVirt CR
+```yaml
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+spec:
+  configuration:
+    network:
```

Comment:
for completeness and ease in case someone uses it as ref
might worth to consider adding please (i do understand otoh that the VEP want to be compact, but it is just a small bit)
```
    developerConfiguration:
      featureGates:
      - NetworkBindingPlugins
```

Reply:
@oshoval The network binding plugins feature was graduated in v1.5.
Please see https://github.com/kubevirt/kubevirt/pull/13314.